x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/user/TransportAuthenticateAction.java

```diff
@@ -35,6 +36,7 @@ public TransportAuthenticateAction(TransportService transportService, ActionFilt
     @Override
     protected void doExecute(Task task, AuthenticateRequest request, ActionListener<AuthenticateResponse> listener) {
         final User runAsUser = securityContext.getUser();
```

Comment:
this is not needed anymore.

Reply:
Hey, can you elaborate? We do still want to return an error if the authenticated user or the run_as user is an internal one. Unless you mean that we can get this from `Authentication#getUser()`

Reply:
Yes, I mean to use `Authentication#getUser()` .